trieste/models/gpflow/utils.py

```diff
@@ -68,7 +68,7 @@ def randomize_hyperparameters(object: gpflow.Module) -> None:
             param.assign(sample)
         elif param.prior is not None:
             # handle constant priors for multi-dimensional parameters
-            if param.prior.batch_shape == param.prior.event_shape == [] and tf.rank(param) == 1:
+            if param.prior.batch_shape == param.prior.event_shape == [] and len(param.shape) == 1:
```

Comment:
maybe add a comment about why this should not be changed to tf.rank? (though now that there's a test at least we'd catch it)